core/src/test/java/kafka/testkit/TestKitNodes.java

```diff
@@ -97,9 +107,12 @@ public Builder setBrokerNodes(int numBrokerNodes, int disksPerBroker) {
                 if (!brokerNodeBuilders.isEmpty()) {
                     nextId = brokerNodeBuilders.lastKey() + 1;
                 }
-                BrokerNode.Builder brokerNodeBuilder = new BrokerNode.Builder()
```

Comment:
Can we evaluate those settings in `build` method? That can simplify code since we don't need to revert the changes (for example, the number of brokers).